src/kafka/kafka.ts

```diff
@@ -0,0 +1,109 @@
+}
+
+export function KafkaConsume(topic: string, consumerConfig?: ConsumerConfig) {
+  function kafkadec<This, Ctx extends DBOSContext, Return>(
+    target: object,
+    propertyKey: string,
+    inDescriptor: TypedPropertyDescriptor<(this: This, ctx: Ctx, ...args: KafkaArgs) => Promise<Return>>
+  ) {
+    const { descriptor, registration } = registerAndWrapFunction(target, propertyKey, inDescriptor);
+    const kafkaRegistration = registration as unknown as KafkaRegistration<This, KafkaArgs, Return>;
+    kafkaRegistration.topic = topic;
+    kafkaRegistration.consumerConfig = consumerConfig;
+
+    return descriptor;
+  }
+  return kafkadec;
+}
+
+/////////////////////////////
+/* Kafka Class Decorators  */
+/////////////////////////////
+
+export interface KafkaDefaults extends RegistrationDefaults {
+  kafkaConfig?: KafkaConfig;
+}
+
+export class KafkaClassRegistration<CT extends { new(...args: unknown[]): object }> extends ClassRegistration<CT> implements KafkaDefaults {
+  kafkaConfig?: KafkaConfig;
+
+  constructor(ctor: CT) {
+    super(ctor);
+  }
+}
+
+export function Kafka(kafkaConfig: KafkaConfig) {
+  function clsdec<T extends { new(...args: unknown[]): object }>(ctor: T) {
+    const clsreg = getOrCreateClassRegistration(ctor) as KafkaClassRegistration<T>;
+    clsreg.kafkaConfig = kafkaConfig;
+  }
+  return clsdec;
+}
+
+////////////////////////
+/* Kafka Management  */
+///////////////////////
+
+export class DBOSKafka{
+  readonly consumers: Consumer[] = [];
+
+  constructor(readonly dbosExec: DBOSExecutor) {}
+
+  async initKafka() {
+    for (const registeredOperation of this.dbosExec.registeredOperations) {
+      const ro = registeredOperation as KafkaRegistration<unknown, unknown[], unknown>;
+      if (ro.topic) {
+        const defaults = ro.defaults as KafkaDefaults;
+        const kafka = new KafkaJS(defaults.kafkaConfig!);
+        const consumerConfig = ro.consumerConfig ?? { groupId: `dbos-kafka-group-${ro.topic}`}
+        const consumer = kafka.consumer(consumerConfig);
+        await consumer.connect()
+        await consumer.subscribe({topic: ro.topic, fromBeginning: true})
+        await consumer.run({
+          eachMessage: async ({ topic, partition, message }) => {
+            // This combination uniquely identifies a message for a given Kafka cluster
+            const workflowUUID = `kafka-unique-id-${topic}-${partition}-${message.offset}`
+            const wfParams = { workflowUUID: workflowUUID };
+            // All operations annotated with Kafka decorators must take in these three arguments
+            const args: KafkaArgs = [topic, partition, message]
+            // We can only guarantee exactly-once-per-message execution of transactions and workflows.
+            if (ro.txnConfig) {
+              // Execute the transaction
+              await this.dbosExec.transaction(ro.registeredFunction as Transaction<unknown[], unknown>, wfParams, ...args);
+            } else if (ro.workflowConfig) {
```

Comment:
Log/throw an error if the config is something else? The current code seems to silently ignore communicators or middleware decorated with a KafkaConsumer decorator.